src/cascadia/TerminalSettingsModel/ActionAndArgs.cpp

```diff
@@ -271,6 +216,49 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return make_self<ActionAndArgs>(action, args);
     }
 
+    Json::Value ActionAndArgs::ToJson(const Model::ActionAndArgs& val)
+    {
+        if (val)
+        {
+            if (!val.Args())
+            {
+                // "command": "copy"
+                const auto shortcutActionIter{ ActionToStringMap.find(val.Action()) };
+                if (shortcutActionIter != ActionToStringMap.end())
+                {
+                    return JsonKey(shortcutActionIter->second);
+                }
+            }
+            else
+            {
+                // "command": { "action": "copy", "singleLine": false }
+                Json::Value result{ Json::ValueType::objectValue };
+
+                // Set the action args, if any
+                const auto actionArgSerializerIter{ argSerializerMap.find(val.Action()) };
+                if (actionArgSerializerIter != argSerializerMap.end())
+                {
+                    auto pfn{ actionArgSerializerIter->second.second };
+                    if (pfn)
+                    {
+                        result = pfn(val.Args());
+                    }
+                }
+
+                // Set the "action" part
+                const auto shortcutActionIter{ ActionToStringMap.find(val.Action()) };
+                result[JsonKey(ActionKey)] = shortcutActionIter == ActionToStringMap.end() ?
+                                                 JsonKey(UnboundKey) :
```

Comment:
this seems weird actually. The code flow suggests that it might accidentally produce a `{action: unbound, copySingleLine: true}` because... we don't _skip it entirely_ when it isn't found in `ActionToStringMap`

Reply:
Refactored it a bit to make it more clear.